src/dataflow/src/source/postgres.rs

```diff
@@ -157,15 +164,39 @@ impl PostgresSourceReader {
                         let a: Datum = row.get(n).into();
                         a
                     }));
-                    snapshot_tx.insert(mz_row).await?;
+                    snapshot_tx.insert(mz_row.clone()).await?;
```

Comment:
This may be related to @petrosagg 's comment below about a different transaction being used to produce and retract.

If there are no failures during snapshotting, the source reports that it is not ready to be queried until it actually is ready:

```
materialize=> select count(*) from t1;
ERROR:  At least one input has no complete timestamps yet: [User(12)]
...
materialize=> select count(*) from t1;
  count  
---------
 9900000
(1 row)
```

But if there has been a failure during the snapshot, the source reports 0 rows instead of an error before it starts reporting the correct number:

```
materialize=> select count(*) from t1;
 count 
-------
     0 <- wrong result
(1 row)
...
materialize=> select count(*) from t1 as of now();
  count  
---------
 9900000
```
This is a consistency violation, as that ```0``` rows is incorrect , there was never a moment with 0 rows on the Pg side.

Reply:
I believe fixing the single transaction issue fixed this bug, but is there a specific test where you were seeing this?

Reply:
I was wrong! I still see this using a single transaction... it might be an issue with the `SourceTransaction` itself? Taking a look now.